src/ucp/tag/rndv.c

```diff
@@ -177,34 +96,51 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_proto_progress_rndv_rtr, (self),
 
 ucs_status_t ucp_tag_send_start_rndv(ucp_request_t *sreq)
 {
+    ucp_ep_h ep = sreq->send.ep;
+    ucp_md_map_t md_map;
     ucs_status_t status;
 
-    ucs_trace_req("starting rndv. sreq: %p. buffer: %p, length: %zu",
-                  sreq, sreq->send.buffer, sreq->send.length);
-    sreq->flags |= UCP_REQUEST_FLAG_RNDV;
-
-    ucp_ep_connect_remote(sreq->send.ep);
+    ucp_trace_req(sreq, "start_rndv buffer %p length %zu", sreq->send.buffer,
+                  sreq->send.length);
+    UCS_PROFILE_REQUEST_EVENT(sreq, "start_rndv", sreq->send.length);
 
-    if (sreq->send.ep->flags & UCP_EP_FLAG_TAG_OFFLOAD_ENABLED) {
+    if (ep->flags & UCP_EP_FLAG_TAG_OFFLOAD_ENABLED) {
         status = ucp_tag_offload_start_rndv(sreq);
         if (status != UCS_OK) {
             return status;
         }
     } else {
-        ucp_request_send_state_reset(sreq, NULL,
-                                     UCP_REQUEST_SEND_PROTO_RNDV_GET);
+        if (UCP_DT_IS_CONTIG(sreq->send.datatype) &&
+            (ep->worker->context->config.ext.rndv_mode != UCP_RNDV_MODE_PUT_ZCOPY))
+        {
+            /* register a contiguous buffer for rma_get */
+            md_map = ucp_ep_config(ep)->key.rma_bw_md_map;
+            status = ucp_request_send_buffer_reg(sreq, md_map);
+            if (status != UCS_OK) {
+                return status;
+            }
+        }
+
+        sreq->send.lane     = ucp_ep_get_am_lane(ep);
```

Comment:
looks like no need to set lane here, can do assert that it is AM lane (for consistency with offload branch)

Reply:
but here we set to am_lan, and in offload.c we expect tag_lane

Reply:
I mean that request lane is set in `ucp_tag_send_req_init()`. No need to set it here again.

Reply:
but ucp_tag_send_req_init() always sets it to tag_lane, and in case of rndv we may need to switch to am_lane.. do you think they would have be the same in this case?

Reply:
if (!ep->flags & UCP_EP_FLAG_TAG_OFFLOAD_ENABLED),  tag_lane should be am_lane

Reply:
ok

Reply:
https://github.com/openucx/ucx/blob/bc72a8f8b140b40ee9b7b4ea533c5e100354c66c/src/ucp/core/ucp_ep.c#L831